src/compiler/binder.ts

```diff
@@ -2608,6 +2608,9 @@ namespace ts {
                 declareSymbol(container.symbol.exports, container.symbol, node, SymbolFlags.ExportStar, SymbolFlags.None);
             }
             else if (isNamespaceExport(node.exportClause)) {
+                // declareSymbol walks up parents to find name text, parent _must_ be set
+                // but won't be set by the normal binder walk until `bindChildren` later on.

```

Comment:
this comment probably isn't needed; we do this in a number of places in the binder and it's an invariant that you'll bump into pretty quickly when writing binder code.

Reply:
Actually we don't. There's only _one_ other instance of `declareSymbol` being called on a child of the current node - it's in special property handling and notably, it does _not_ set the parent, so very likely may actually be indicative of a similar bug.